lib/api/apiUtils/object/checkUserMetadataSize.js

```diff
@@ -0,0 +1,38 @@
+const { maximumMetaHeadersSize,
+    invalidObjectUserMetadataHeader } = require('../../../../constants');
+
+/**
+ * Checks the size of the user metadata in the object metadata and removes
+ * them from the response if the size of the user metadata is larger than
+ * the maximum size allowed. A custome metadata key is added to the response
+ * with the number of user metadata keys not returned as its value
+ * @param {object} responseMetadata - response metadata
+ * @return {object} responseMetaHeaders headers with object metadata to include
+ * in response to client
+ */
+function checkUserMetadataSize(responseMetadata) {
+    let md = {};
+    let userMetadataSize = 0;
+    md = JSON.parse(JSON.stringify(responseMetadata));
```

Comment:
I think this can be a significant CPU eater when the metadata is not tiny. On the other hand, it seems you would only need to copy if the usermd size is above the limit, hence I recommend to do this "copy" down in the `if (userMetadataSize > maximumMetaHeadersSize)` block.

Reply:
Or even less costly would be to simply reassign headers in the new object without going through JSON serialization tricks.